src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java

```diff
@@ -25,7 +29,10 @@ public synchronized void setCurrentDriver(Job2dDriver driver) {
         container.remove(currentDriver);
         container.add(driver);
         currentDriver = driver;
-
+        DriverCounterVisitor counter = new DriverCounterVisitor();
+        VisitableDriver dr = (VisitableDriver) driver;
+        dr.accept(counter);
+        logger.info(counter.toString());
```

Comment:
Do not pollute a code of a method with unrelated and optional responsibilities. 
Moreover do not use visitor toString, but provide appropriate getter.

Drivers counter can be added as DriverManager observer or as an additional test in application menu.

Reply:
Done